addons/account/models/account_move.py

```diff
@@ -871,27 +873,32 @@ def _compute_name(self):
             if move.state == 'cancel':
                 continue
 
-            move_has_name = move.name and move.name != '/'
-            if move_has_name or move.state != 'posted':
-                if not move.posted_before and not move._sequence_matches_date():
-                    if move._get_last_sequence():
-                        # The name does not match the date and the move is not the first in the period:
-                        # Reset to draft
-                        move.name = False
-                        continue
-                else:
-                    if move_has_name and move.posted_before or not move_has_name and move._get_last_sequence():
-                        # The move either
-                        # - has a name and was posted before, or
-                        # - doesn't have a name, but is not the first in the period
-                        # so we don't recompute the name
-                        continue
-            if move.date and (not move_has_name or not move._sequence_matches_date()):
+            move_has_name = move.name and move.name != "/"
+            if move_has_name or move.state != "posted":
+                if move_has_name and move.posted_before or move.state != "posted":
+                    # The move either
+                    # - has a name and was posted before, or
+                    # - doesn't have a name, but is not requiring one (draft, etc.)
+                    # so we don't recompute the name
+                    continue
+                elif move_has_name and not move.posted_before and not move._sequence_matches_date():
+                    # The name does not match the date and the move is not the first in the period:
+                    # Reset to draft
+                    move.name = False
+                    continue
+            if move.date and not move_has_name and move.state == "posted":
                 move._set_next_sequence()
 
-        self.filtered(lambda m: not m.name and not move.quick_edit_mode).name = '/'
         self._inverse_name()
 
+    @api.depends('date', 'journal_id', 'move_type', 'name', 'posted_before', 'sequence_number', 'sequence_prefix', 'state')
+    def _compute_name_placeholder_form(self):
+        self.name_placeholder_form = False
+        for record in self.filtered(lambda move: not move.name or not move.posted_before):
+            if not (record._get_last_sequence() or record.state != "draft"):
```

Comment:
Why are you filtering in 2 different places?
```suggestion
        for record in self.filtered(lambda move: 
            (not move.name or not move.posted_before)
            and not move._get_last_sequence() 
            and move.state == "draft"
        ):
```

```suggestion
        for move in self: 
            if (
                (not move.name or not move.posted_before)
                and not move._get_last_sequence() 
                and move.state == "draft"
            ):
```

Reply:
If I'm reading your condition right, the `state` has to be `draft`. So `posted_before` cannot be `True`

Reply:
Reseting to draft a posted move exactly does that, no ? state == "draft" and "posted_before" == True ?